recipes-kernel/linux/files/0018-add-the-sysfw-ABI3.X-support.patch

```diff
@@ -0,0 +1,74 @@
+From eee32b0cfc5eaa8419b5f73770a34bd2aadd9318 Mon Sep 17 00:00:00 2001
+From: Chao Zeng <chao.zeng@siemens.com>
+Date: Tue, 15 Dec 2020 14:33:34 +0800
+Subject: [PATCH 18/18] add the sysfw ABI3.X support
+
+Signed-off-by: Chao Zeng <chao.zeng@siemens.com>
+---
+ arch/arm64/boot/dts/siemens/Makefile             |  2 +-
+ .../boot/dts/siemens/iot2050-advanced-abi3.x.dts | 16 ++++++++++++++++
+ .../boot/dts/siemens/iot2050-basic-abi3.x.dts    | 16 ++++++++++++++++
```

Comment:
iot2050-{basic,advanced}.dts should imply abi3.x. Use special names only for the legacy dtbs. Just like we do for the basic dtb (siemens/iot2050-basic-oldfw.dtb).

Reply:
Question is that when using transition kernel with the released old u-boot, the dtb file is recgnized to "iot2050-basic.dtb" or "iot2050-advanced.dtb", which is using the new 3.x abi.

Reply:
We need to following logic in the wks file:

if env exists fdtfile -> new firmware, new sysfw
else if board_name = advanced -> setenv fdtfile advanced-oldfw.dtb
​else setenv fdtfile basic-oldfw.dtb

That's the existing one plus detection of legacy fw on advanced devices.

Reply:
...and all this legacy stuff will then be removed without traces in the next-but-one image release. Think this through from that perspective.

Reply:
could the environment `fdtfile`  act as a flag to distinguish new firmware , new sysfw ? 
I have seen the the fdtfile has been set in the upstream u-boot according to the board type. am I right?

Reply:
Exactly, see also f5c53afb9b82f748a793a8620bee475c1e5c314b.

Reply:
Ok I get it. thanks

Reply:
But I have another question. I see there is a iot2050-basic-oldfw.dts, it is compat the V01.0.0.1 . and could it able to Isolate this file?

Reply:
Sorry, I do not understand the question yet.